[Bug] Fail to build test-related target (swift-snapshot-testing/SnapshotTesting)

### What happened?

Fail to build swift-snapshot-testing/SnapshotTesting (tried in the Kickstarter project).

<img width="1118" alt="Image" src="https://github.com/user-attachments/assets/c858465a-858e-4053-94bc-fd0c92702199" />

### Anything else?

Build errors (from `swift build...`)

```
[0/1] Planning build
    Building for debugging...
    [0/1] Write swift-version-5BDAB9E9C0126B9D.txt
    warning: save unknown driver flag -alias-module-names-in-module-interface as additional swift-frontend flag
    [2/5] Compiling SnapshotTesting AssertSnapshot.swift
    /Users/thuyen/projects/kickstarter/xccache/packages/umbrella/.build/checkouts/swift-snapshot-testing/Sources/SnapshotTesting/AssertSnapshot.swift:463:24: error: type 'XCTContext' has no member 'runActivity'
    461 |             !isSwiftTesting
    462 |           {
    463 |             XCTContext.runActivity(named: "Attached Failure Diff") { activity in
        |                        `- error: type 'XCTContext' has no member 'runActivity'
    464 |               attachments.forEach {
    465 |                 activity.add($0)
    [3/5] Compiling SnapshotTesting Deprecations.swift
    /Users/thuyen/projects/kickstarter/xccache/packages/umbrella/.build/checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Internal/Deprecations.swift:141:22: error: type 'XCTContext' has no member 'runActivity'
    139 |       #if !os(Linux) && !os(Android) && !os(Windows)
    140 |         if ProcessInfo.processInfo.environment.keys.contains("__XCODE_BUILT_PRODUCTS_DIR_PATHS") {
    141 |           XCTContext.runActivity(named: "Attached Failure Diff") { activity in
        |                      `- error: type 'XCTContext' has no member 'runActivity'
    142 |             attachments.forEach {
    143 |               activity.add($0)
    [4/5] Compiling SnapshotTesting RecordIssue.swift
    /Users/thuyen/projects/kickstarter/xccache/packages/umbrella/.build/checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Internal/RecordIssue.swift:38:5: error: cannot find 'XCTFail' in scope
    36 |     }
    37 |   #else
    38 |     XCTFail(message(), file: filePath, line: line)
       |     `- error: cannot find 'XCTFail' in scope
    39 |   #endif
    40 | }
    
    /Applications/Xcode-16.2.0.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks/XCTest.framework/Headers/XCTestAssertions.h:45:9: note: macro 'XCTFail' unavailable: function like macros not supported
     43 |  * @param ... An optional supplementary description of the failure. A literal NSString, optionally with string format specifiers. This parameter can be completely omitted.
     44 | */
     45 | #define XCTFail(...) \
        |         `- note: macro 'XCTFail' unavailable: function like macros not supported
     46 |     _XCTPrimitiveFail(nil, __VA_ARGS__)
     47 | 
    [5/5] Emitting module SnapshotTesting
    [ERROR] Failed to build target: SnapshotTesting. Error: Command 'swift build --configuration debug --triple arm64-apple-ios-simulator --package-path /Users/thuyen/projects/kickstarter/xccache/packages/umbrella --target SnapshotTesting --sdk /Applications/Xcode-16.2.0.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk -Xswiftc -enable-library-evolution -Xswiftc -alias-module-names-in-module-interface -Xswiftc -emit-module-interface -Xswiftc -no-verify-emitted-module-interface' failed with status: 1
```

--- a/examples/EX.xcodeproj/project.pbxproj
+++ b/examples/EX.xcodeproj/project.pbxproj
@@ -36,9 +36,9 @@
 /* End PBXCopyFilesBuildPhase section */
 
 /* Begin PBXFileReference section */
-		1CBE8AAACA849C519746C865 /* xccache.yml */ = {isa = PBXFileReference; includeInIndex = 1; path = xccache.yml; sourceTree = "<group>"; };
+		1CBE8AAACA849C519746C865 /* xccache.yml */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.yaml; path = xccache.yml; sourceTree = "<group>"; };
 		3B5B411E32E4429CC9EDBC87 /* Package.swift */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = sourcecode.swift; name = Package.swift; path = xccache/packages/umbrella/Package.swift; sourceTree = "<group>"; };
-		D1B6F890297CAD42EA43AFB3 /* xccache.lock */ = {isa = PBXFileReference; includeInIndex = 1; path = xccache.lock; sourceTree = "<group>"; };
+		D1B6F890297CAD42EA43AFB3 /* xccache.lock */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text; path = xccache.lock; sourceTree = "<group>"; };
 		F577C62F2D96971200C83C96 /* EX.app */ = {isa = PBXFileReference; explicitFileType = wrapper.application; includeInIndex = 0; path = EX.app; sourceTree = BUILT_PRODUCTS_DIR; };
 		F577C6A82D96A3BD00C83C96 /* config.xcconfig */ = {isa = PBXFileReference; lastKnownFileType = text.xcconfig; path = config.xcconfig; sourceTree = "<group>"; };
 		F5AF0F9C2DAE09EF00AB812D /* EXTests.xctest */ = {isa = PBXFileReference; explicitFileType = wrapper.cfbundle; includeInIndex = 0; path = EXTests.xctest; sourceTree = BUILT_PRODUCTS_DIR; };

--- a/examples/EXTests/SPMPlayground.swift
+++ b/examples/EXTests/SPMPlayground.swift
@@ -1,0 +1,10 @@
+import DebugKit
+import ResourceKit
+import TestKit
+
+func playground() {
+  print(DebugKit.self)      // DebugKit
+  print(ResourceKit.self)   // ResourceKit
+  print(TestKit.self)       // TestKit
+  print(BaseTestCase.self)  // TestKit
+}

--- a/examples/LocalPackages/core-utils/Package.swift
+++ b/examples/LocalPackages/core-utils/Package.swift
@@ -9,6 +9,7 @@ let package = Package(
     .library(name: "Swizzler", targets: ["Swizzler"]),
     .library(name: "ResourceKit", targets: ["ResourceKit"]),
     .library(name: "DebugKit", targets: ["DebugKit"]),
+    .library(name: "TestKit", targets: ["TestKit"]),
   ],
   dependencies: [
     .package(url: "https://github.com/SwiftyBeaver/SwiftyBeaver.git", .upToNextMajor(from: "2.1.1")),
@@ -45,5 +46,6 @@ let package = Package(
       name: "CoreUtils-Wrapper",
       path: "Sources/Core"
     ),
+    .target(name: "TestKit"),
   ]
 )

--- a/examples/LocalPackages/core-utils/Sources/TestKit/TestKit.swift
+++ b/examples/LocalPackages/core-utils/Sources/TestKit/TestKit.swift
@@ -1,0 +1,7 @@
+import Testing
+import XCTest
+
+@objc public class TestKit: NSObject { }
+
+public class BaseTestCase: XCTestCase { }
+public protocol BaseTrait: Trait { }

--- a/examples/xccache.lock
+++ b/examples/xccache.lock
@@ -77,7 +77,9 @@
         "SnapKit/SnapKit-Dynamic",
         "KingfisherWebP/KingfisherWebP"
       ],
-      "EXTests": []
+      "EXTests": [
+        "core-utils/TestKit"
+      ]
     }
   }
 }

--- a/lib/xccache/framework/slice.rb
+++ b/lib/xccache/framework/slice.rb
@@ -28,6 +28,7 @@ module XCCache
           cmd << "-Xswiftc" << "-alias-module-names-in-module-interface"
           cmd << "-Xswiftc" << "-emit-module-interface"
           cmd << "-Xswiftc" << "-no-verify-emitted-module-interface"
+          sdk.swiftc_args.each { |arg| cmd << "-Xswiftc" << arg }
           Sh.run(cmd, suppress_err: /(dependency '.*' is not used by any target|unable to create symbolic link)/)
           create_framework
         end

--- a/lib/xccache/swift/sdk.rb
+++ b/lib/xccache/swift/sdk.rb
@@ -25,7 +25,23 @@ module XCCache
       end
 
       def sdk_path
+        # rubocop:disable Layout/LineLength
+        # /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk
+        # rubocop:enable Layout/LineLength
         @sdk_path ||= Pathname(Sh.capture_output("xcrun --sdk #{name} --show-sdk-path")).realpath
+      end
+
+      def sdk_platform_developer_path
+        @sdk_platform_developer_path ||= sdk_path.parent.parent # iPhoneSimulator.platform/Developer
+      end
+
+      def swiftc_args
+        developer_library_frameworks_path = sdk_platform_developer_path / "Library" / "Frameworks"
+        developer_usr_lib_path = sdk_platform_developer_path / "usr" / "lib"
+        [
+          "-F#{developer_library_frameworks_path}",
+          "-I#{developer_usr_lib_path}",
+        ]
       end
     end
   end

